src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementEventNormalizationService.cs

```diff
@@ -92,33 +94,53 @@ private async Task StartConsumer(ISourceBlock<EventData> producer, IAsyncCollect
                         foreach (var measurement in _contentTemplate.GetMeasurements(token))
                         {
                             measurement.IngestionTimeUtc = evt.SystemProperties.EnqueuedTimeUtc;
-                            await collector.AddAsync(measurement).ConfigureAwait(false);
-
-                            _log.LogMetric(
-                                IomtMetrics.NormalizedEvent(),
-                                1);
+                            createdMeasurements.Add(measurement);
                         }
                     }
-                    catch (OperationCanceledException)
-                    {
-                        cts.Cancel();
-                        throw;
-                    }
-#pragma warning disable CA1031
                     catch (Exception ex)
                     {
                         if (await errorConsumer(ex, evt).ConfigureAwait(false))
                         {
                             exceptions.Add(ex);
                         }
                     }
-#pragma warning restore CA1031
+
+                    return createdMeasurements;
+                });
+
+            var asyncCollectorConsumer = new ActionBlock<IMeasurement[]>(
+                async measurements =>
+                {
+                    try
+                    {
+                        await collector.AddAsync(measurements, cts.Token).ConfigureAwait(false);
+                        _log.LogMetric(
```

Comment:
>LogMetric [](http://example.com/codeflow?start=29&length=9)

This is optional but I was thinking if we could have the destination partition in scope here it would be helpful in diagnosising if we have a hot partition.  This might be better for another PR/work item if we want to do this because I feel the refactor would be extensive but at least wanted to discuss.

Reply:
I actually really like that idea, and logging it as a metric. At least we'd able to keep track of how we are distributing partition keys in our logic. We'd still need to leverage the work that Pallavi's has done to know how the EventHub  is actually distributing them across the EventHub but have the information here would help too.

Reply:
I actually did include the partitionId for this event already. We may just need to pull latest - https://github.com/microsoft/iomt-fhir/blob/master/src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementEventNormalizationService.cs#:~:text=IomtMetrics.NormalizedEvent(partitionId)%2C

Reply:
Thanks Pallavi.  I think with your changes this will be the source partition id and not the destination partition but after discussing with Rob Event Hub still hashes the partition value we assign to get the true underlying partition so we don't know the actual destintion partiton at this point.

While knowing the destination partition the normalized event is going to would be helpful I don't know if we could realistically do that at this point.  We could track this on the FHIR conversion side but we would only know we have a hot partition if we have been able to convert enough data to FHIR.  Still useful for analytics but it might not help diagnosis a hot partition in real time.  Open to further thoughts.